src/mesh/coordinates.cxx

```diff
@@ -46,7 +172,10 @@ Coordinates::Coordinates(Mesh *mesh)
       G1_23(mesh), G2_11(mesh), G2_22(mesh), G2_33(mesh), G2_12(mesh), G2_13(mesh),
       G2_23(mesh), G3_11(mesh), G3_22(mesh), G3_33(mesh), G3_12(mesh), G3_13(mesh),
       G3_23(mesh), G1(mesh), G2(mesh), G3(mesh), ShiftTorsion(mesh),
-      IntShiftTorsion(mesh), localmesh(mesh), location(CELL_CENTRE) {
+      IntShiftTorsion(mesh), localmesh(mesh), options(options_in), location(CELL_CENTRE) {
+
+  std::string paralleltransform = options["paralleltransform"].withDefault(std::string("identity"));
+  bool extrapolate_periodic_boundary = (paralleltransform == "identity");
```

Comment:
Is this extrapolation to avoid communication? If the domain is periodic, surely we don't need to extrapolate?

Reply:
I'm a bit nervous of code that has to know what the specific type of an object is -- ideally we'd do the same thing for all types or at least ask the instance if it can do an operation if it only makes sense to do it for certain cases (rather than it's possible for one case but not for another due to a lack of information, but would make sense for both if we had all the information available).

Reply:
My choice of variable names was bad. I should have called this something like `extrapolate_branch_cut`. The point is that at the poloidal branch cut on closed field lines, with a field-aligned grid the integrated shear `I` isn't continuous, so some of the metric components are also not continuous. But the discontinuity is a branch cut like \theta=2\pi->0 so gradients near the branch cut should not be affected.

If we communicate in the field-aligned case, the guard cells at the branch cut boundary come from the wrong side of the branch cut. I think the most straightforward way to get around this is to extrapolate into the guard cells at the branch cut. It would be more accurate not to extrapolate, but we'd need to save some extra information in the grid file and that doesn't seem worthwhile as if you need accuracy near the branch cut you shouldn't be using the identity transform anyway.

Reply:
@d7919 I agree this test should be a method of `ParallelTransform` rather than reading the option here. I was trying to avoid the need to add to the interface, but that was the wrong choice here.

Reply:
I'm still a bit confused -- which branch cut are you talking about?

Presumably this only applies in the BOUT++ tokamak coordinates, and not simple geometries like slab?

Would a simpler solution be to just generate quantities like `I` in the guard cells?

Reply:
It's the branch cut where twist-shift is applied.

Only really applies in tokamak coordinates, although a sheared slab would have a similar issue.

Generating `I` in the guard cells would be a solution, but to do it generally we'd have to change the structure of the grid files, because if there's a private flux region, then in the global, cartesian grid the PFRs are adjacent to the y-boundaries of the core region
(see https://bout-dev.readthedocs.io/en/latest/user_docs/input_grids.html#advanced). So it's not straightforward to add guard cells to the grid input that could store the extra pieces of `I`. [We'd also have to add back some functionality to generate metric components from `I`, `Bxy`, etc., but that shouldn't be a problem.]

I got hit by this because I'm trying to update a test of the differential operators that includes non-trivial metrics and varying grid spacings. I'm using a large-aspect-ratio tokamak-like geometry (from analytic expressions). Possibly I should give up on that and just generate some analytical metrics that are non-constant but periodic to test the field-aligned case, and only test tokamak geometry with ShiftedMetric. In that case I think we should add a warning somewhere that field-aligned simulations in tokamak geometry may have issues, especially as `paralleltransform = identity` is the default.

Reply:
I guess the identity transform also has problems with x-derivatives for tokamak geometry as well, so you're likely to run into multiple problems. I'm not sure where the best place for such a warning is -- the manual somewhere?

Providing some functionality to automatically set-up tokamak geometry that defaults the transform to shifted would be nice too.